### PR TITLE
fix(tests): update to mocha 5

### LINKED ISSUE #2531: Update mocha?
I tried to do this as part of #2530, but the latest mocha hangs if you have running async listeners. It turns out we have a lot of those.

There are some [instructions here](https://gist.github.com/boneskull/7fe75b63d613fa940db7ec990a5f5843) for debugging them. I tried that locally and the output was enough to overflow my terminal buffer, so I won't paste the whole thing here. A lot of them seemed to be poolee-related and a lot seemed to be `lib/server.js`-related. Hopefully it's straightforward to fix them *en masse* (ish).

This is a nice to have, but not the end of the world if we can't make it work.

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #2531

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
LGTM, though when I try to run the tests locally I get a bunch of problems:

> (node:67251) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 39)
(node:67251) UnhandledPromiseRejectionWarning: Error: listen EADDRINUSE 127.0.0.1:9010
    at Object._errnoException (util.js:1022:11)
    at _exceptionWithHostPort (util.js:1044:20)
    at Server.setupListenHandle [as _listen2] (net.js:1367:14)
    at listenInCluster (net.js:1408:12)
    at GetAddrInfoReqWrap.doListen [as callback] (net.js:1517:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:97:10)****

as well as:

> Unhandled rejection (<{"statusCode":400,"code":42,"message":...>, no stack trace)

**Comment by vladikoff** [comment]
> node:67251) UnhandledPromiseRejectionWarning: Error: listen EADDRINUSE 127.0.0.1:9010

I think this happens because you have local servers running, those need to be stopped for tests to pass

> Unhandled rejection (<{"statusCode":400,"code":42,"message":...>, no stack trace)

this is a promise problem, seems like the only way to fix this is to rewrite the test a bit. This problem is in current `master` also I think

**Comment by shane-tomlinson** [comment]
> I think this happens because you have local servers running, those need to be stopped for tests to pass

When I do this, I run into:

> test.CRITICAL: uncaughtException Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
    at Object._errnoException (util.js:1022:11)
    at _exceptionWithHostPort (util.js:1044:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1198:14)

redis required to be running for the tests to pass?

**Comment by shane-tomlinson** [comment]
> redis required to be running for the tests to pass?

Yes.

File: scripts/test-local.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ if [ -z "$GLOB" ]; then
   GLOB="test/local test/remote"
 fi
 
-DEFAULT_ARGS="-R dot --recursive --timeout 5000"
+DEFAULT_ARGS="-R dot --recursive --timeout 5000 --exit"
```
Comment (vladikoff) [inline scripts/test-local.sh]: I got this to work without `--exit` by adding `redis.close()` in certain places + some other stuffs. 

Using this:
https://gist.github.com/boneskull/7fe75b63d613fa940db7ec990a5f5843

**Pull request merged**